### ISSUE TITLE
No Audio Through HDMI Possible

### ISSUE THREAD

**Issue by eOSfreak** [elementary/switchboard-plug-sound#65]
Hey eOS developers,
unfortunately I can't choose HDMI as sound output method. I tried another laptop with the same cable and it's working so the issue seems to be connected to the release of eOS Juno Beta.

**Comment by alexgleason** [comment]
I'm testing Juno beta 2 as a home theater OS. HDMI output doesn't appear as an option in the output dialog. It works fine on Ubuntu 14.04 with the same hardware.

Supposedly there's [a known issue](https://askubuntu.com/questions/1053076/18-04-sound-through-hdmi) with 18.04 where HDMI audio is lost after returning from sleep. In my case, even rebooting the machine has no effect.

**Comment by alexgleason** [comment]
For now I've worked around the problem.

```
sudo apt install pavucontrol
```

Then launch pavucontrol from the applications menu, go to the "Configuration" tab, and select "Digital Stereo (HDMI) Output" from the "Profile" dropdown under "Built-in Audio." This makes sound come out of the speakers.

**Comment by alexgleason** [comment]
So, I noticed the countdown which is very exciting! Though I'm wondering if these issues with sound have been fixed as I'm concerned the change here is going to be a major pain-point for users.

I've noticed each year users are getting more and more impatient, saying "elementary is dead" etc, and elementary has become exponentially faster at shipping releases. But if you give them what they want I have a feeling they may just complain even more, lol. Just my 2 cents, appreciate your work either way.

**Comment by peteruithoven** [comment]
I've been able to reproduce this issue and I'm able to get HDMI working using pavucontrol.

I placed some logs in the `PulseAudioManager` and `OutputPanel` files and they don't seem to get info about the HDMI port. There also isn't a function triggered as a response to connecting a HDMI device.

**Comment by peteruithoven** [comment]
I'm afraid this limitation is a result of how we implemented the UI on top of the PulseAudio API, following their model: https://github.com/elementary/switchboard-plug-sound/issues/80

**Comment by msssouza** [comment]
Thanks. Pavucontrol saved my day.

**Comment by Marcos-Costa** [comment]
Thanks 

> For now I've worked around the problem.
> 
> ```
> sudo apt install pavucontrol
> ```
> 
> Then launch pavucontrol from the applications menu, go to the "Configuration" tab, and select "Digital Stereo (HDMI) Output" from the "Profile" dropdown under "Built-in Audio." This makes sound come out of the speakers.
> 
> This confirms it's a UI issue and not a kernel problem.

Thanks, that worked for me, I'm having the same issue.

**Comment by Gabriel-p** [comment]
Still an issue 6 months later after the "stable" release, sadly.

**Comment by Gabriel-p** [comment]
Still an issue 6 months later of being reported and after the "stable" release, sadly.

**Comment by Gabriel-p** [comment]
Still an issue after 6 months of being reported and even after the "stable" version was released, sadly.

**Comment by leoherzog** [comment]
Also having this issue. `pavucontrol` helped, but now I'm unable to switch back.

**Comment by HansCronau** [comment]
Also having this issue since Juno.

@leoherzog For a second I too thought I couldn't switch back. It then turned out I was in the tab "Output Devices", while the profile switching is done in the tab "Configuration". Could that too have been the case for you?

**Comment by leoherzog** [comment]
@HansCronau After a mishmash of restarts and changing settings, I got it working again. Worried about the day I'll have to use HDMI audio again though!

**Comment by peteruithoven** [comment]
Hey people you could help out by testing the following PR: 
https://github.com/elementary/switchboard-plug-sound/pull/83
It's a big change and requires some knowledge of PulseAudio so the review is taking quite a while.

**Comment by naaando** [comment]
@peteruithoven I've been testing since you began working on this, working great so far for my user case(switching onboard audio to HDMI)

**Comment by kbondarev** [comment]
@peteruithoven, I'm having the same issue with the HDMI and would love to help and test that PR.
Could you point me to some docs explaining how to build and install it?

**Comment by peteruithoven** [comment]
@kbondarev The readme always contain information on how to build: https://github.com/elementary/switchboard-plug-sound
After building and installing it's a matter of re-opening the System Settings application.

**Comment by kbondarev** [comment]
@peteruithoven, thanks!
I have installed it and it works great for now, the moment I plug in the HDMI it automatically switched the output to it and I can see it as an output option in the settings.
If I'll bump into any issues, I will report here :)

**Comment by HansCronau** [comment]
@peteruithoven Works perfectly. Many thanks.

**Comment by HansCronau** [comment]
Update: when connecting my bluetooth headphones they gave no sound. The settings indicated that "Headset" was selected as output, but it wasn't until I manually switched to "Speakers" and back to "Headset" that they actually started producing sound. I tried to reproduce without success: no matter what output was selected before disconnecting ("Speakers" or "Headset"), the headset would be selected automatically _and_ produce sound after reconnecting as (I) expected. Could it be that this happens only the very first time, when the headset output hasn't been manually selected before?
